### ISSUE TITLE
OpenAPI should list a 400 error code on certain conditions

### ISSUE THREAD

**Issue by pilhuhn** [quarkusio/quarkus#27376]
### Description

Suppose you have

```
    @POST
    @Path("/example")
    public void addBar(MyBean myBean) {
        System.out.println(myBean.bar);
    }
```

Then OpenAPI should list a 400 code in the responses section (unless the user explicitly provides one), as RestEasy will reject the request when the passed data can't be de-serialized into `myBean`.

If a `@Valid` annotation is present, the same applies, but it means that the constraints inside the bean are checked as well. E.g. 

```
public class MyBean {

    @NotNull @Length(max = 6, min = 2)
    String bar;

```

Here the component.schema should list the min/max length and also the 400 error should be listed if not done before 

Related to #27375

**Comment by fercomunello** [comment]
Did you tried to add the Valid annotation to your endpoint? This annotation will cascade the validation to your bean:
```java
@POST
@Path("/example")
public void addBar(@Valid MyBean myBean) {
    System.out.println(myBean.bar);
}
```

**Comment by pilhuhn** [comment]
@fercomunello Yes, but still no 400 response automatically created, which this one is about :)

**Comment by phillip-kruger** [comment]
These are all good suggestions. Thanks. I'll add them to my list of todos :)
